src/portfft/defines.hpp

```diff
@@ -48,6 +49,14 @@ namespace portfft {
 using Idx = std::int32_t;
 using IdxGlobal = std::int64_t;
 
+/**
+ * An array of 2 arrays containing N elements of Type, containing strides (s) and offset (o) for a view
+ * @tparam Type Type of elements
+ * @tparam N Number of elements in each of the two arrays
+ */
+template <typename Type, Idx N>
+using so_array = std::array<std::array<Type, N>, 2>;
```

Comment:
I am bit wary of using `structs` on device. It's not that I did not think of using `structs` but it's rather something I have avoided on purpose.
 
As I have been reading recently, best  practices typically dictate to avoid using struct (and I am not talking about passing a struct pointer, or SOA vs AOS etc, but rather creating local struct objects) as they are known to spill more.

whenever I build for the cuda backend , I pass the `--warn-on-local-memory-usage` flags to ptxas, and not just warn on spills. Nevertheless, I switched to using structs, instead of an array of array, and compiled again with the above mentioned flags, and saw larger spills to local memory. Though I am aware this spillage numbers might not be exact because of spec-constant issues. I would rather stick to the version giving me lesser spills even today.

Also, we might want to revisit the use of structs in our views.

Reply:
Compiler should be able to break up structs. If it does not we can report a bug.

Reply:
Using structs now, though we would need to look into it's performance at some point